### PR TITLE
improvement(VCalendar): show week numbers in month-view

### LINKED ISSUE #9117: [Feature Request] Show week numbers in v-calendar
### Problem to solve
Currently there is no way to see week numbers in v-calendar.

### Proposed solution
A week number option that can be filled out to add week numbers to the calendar view. For example in the leftmost column, above hours.

The config property should be able to specify how week number 1 is found. E.g. the first week with a monday is week 1 or the first 4 day week is week 1.

This was fixed for date-picker, see https://github.com/vuetifyjs/vuetify/issues/5077 for reference.

### PR DESCRIPTION
**Pull request by abbierwolf**
re #9117

## Description
Resolves #9117 partly only for month-view.

## Motivation and Context
Pull-request #9974 made it possible to manually add a week number using a slot in week/day-view. This pull-request adds a boolean flag to show the week numbers in front of the week.

https://github.com/vuetifyjs/vuetify/issues/9117

## How Has This Been Tested?
unit | visually

## Markup:
<details>

```vue
<template>
  <v-container>
    <!--  -->
    <v-sheet height="600px">
      <v-calendar
        show-week
        start="2017-01-01"
        weekdays="1, 2, 3, 4, 5, 6, 0"
      ></v-calendar>
    </v-sheet>
  </v-container>
</template>

<script>
  export default {
    data: () => ({
    //
    }),
  }
</script>
```
</details>

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:
- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by jacekkarczmarczyk** [comment]
As I understand this PR waits first for https://github.com/vuetifyjs/vuetify/pull/10926?

**Comment by abbierwolf** [comment]
@jacekkarczmarczyk Correct if #10926 is merged the weekNumber function can be re-used here.

**Comment by abbierwolf** [comment]
@jacekkarczmarczyk Could you review the code?

File: packages/vuetify/src/components/VCalendar/VCalendarWeekly.ts [hunk 1]
```diff
@@ -108,17 +117,40 @@ export default CalendarBase.extend({
...
+        determineDay.day,
+        this.parsedWeekdays[0],
+        parseInt(this.localeFirstDayOfYear)
+      )
+    },
+    genWeekNumber (weekNumber: number) {
+      return this.$createElement('div', {
+        staticClass: 'v-calendar-weekly__weeknumber',
+      }, [
+        this.$createElement('small', { }, String(weekNumber).padStart(2, '0')),
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VCalendar/VCalendarWeekly.ts:152]:
```suggestion
        this.$createElement('small', String(weekNumber).padStart(2, '0')),
```

File: packages/vuetify/src/components/VCalendar/__tests__/VCalendar.spec.ts [hunk 2]
```diff
@@ -78,6 +78,24 @@ describe('VCalendar', () => {
...
   })
 
+  it('should render month view with show-week', async () => {
+    const wrapper = mountFunction({
+      propsData: {
+        type: 'month',
+        start: '2018-01-29',
+        end: '2018-02-04',
+        now: '2019-02-17',
+        'show-week': true,
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VCalendar/__tests__/VCalendar.spec.ts:88]:
```suggestion
        'showWeek': true,
```

File: packages/vuetify/src/components/VCalendar/util/props.ts [hunk 3]
```diff
@@ -98,6 +102,10 @@ export default {
       type: Boolean,
       default: true,
     },
+    showWeek: {
+      type: Boolean,
+      default: false,
+    },
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VCalendar/util/props.ts:108]:
```suggestion
    showWeek: Boolean,
```

File: packages/vuetify/src/components/VCalendar/VCalendarWeekly.sass [hunk 4]
```diff
@@ -47,13 +54,32 @@
   text-transform: uppercase
   white-space: nowrap
 
+.v-calendar-weekly__head-weeknumber
+  position: relative
+  border-top-left-radius: 4px
+  border-top-right-radius: 4px
+  flex: 0 0 24px
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VCalendar/VCalendarWeekly.sass:61]: move 24px to sass var (and reuse it in `.v-calendar-weekly__weeknumber`

File: packages/vuetify/src/components/VCalendar/VCalendarWeekly.sass [hunk 5]
```diff
@@ -47,13 +54,32 @@
...
+  top: 0.2em
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VCalendar/VCalendarWeekly.sass:62]: what is this for? it causes the space at the top, is it expected?
image

could be a sass var too, along with border radius

File: packages/vuetify/src/components/VCalendar/VCalendarWeekly.sass [hunk 6]
```diff
@@ -47,13 +54,32 @@
...
   // https://github.com/vuetifyjs/vuetify/issues/8319
   min-height: 0
 
+.v-calendar-weekly__weeknumber
+  display: flex
+  flex: 0 0 24px
+  font-weight: 500
+  height: unset
+  min-height: 0
+  padding-top: 8px
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VCalendar/VCalendarWeekly.sass:77]: sass var

File: packages/vuetify/src/components/VCalendar/__tests__/VCalendar.spec.ts [hunk 7]
```diff
@@ -78,6 +78,24 @@ describe('VCalendar', () => {
...
+      },
+      methods: {
+        getNow: () => parseDate(new Date('2019-02-17')),
+      },
+    })
+
+    expect(wrapper.classes('v-calendar-monthly')).toBeTruthy()
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VCalendar/__tests__/VCalendar.spec.ts:95]: this doesn't seem related

**Comment by jacekkarczmarczyk** [comment]
how about making it more similar to week numbers in date picker?
image

lighter grey color and maybe same background?

**Comment by abbierwolf** [comment]
@jacekkarczmarczyk Could be a possibility but for this change I looked at the gmail calendar

**Comment by jacekkarczmarczyk** [comment]
ok, actually week numbers in date picker also looks different in google calendar that in vuetify

File: packages/vuetify/src/components/VCalendar/VCalendarWeekly.ts [hunk 8]
```diff
@@ -108,17 +117,40 @@ export default CalendarBase.extend({
...
+        determineDay.day,
+        this.parsedWeekdays[0],
+        parseInt(this.localeFirstDayOfYear)
+      )
+    },
+    genWeekNumber (weekNumber: number) {
+      return this.$createElement('div', {
+        staticClass: 'v-calendar-weekly__weeknumber',
+      }, [
+        this.$createElement('small', String(weekNumber).padStart(2, '0')),
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VCalendar/VCalendarWeekly.ts:152]: i don't think we should use pad here (google doesn't use it)

**Comment by abbierwolf** [comment]
@jacekkarczmarczyk Any idea when review can be completed?

File: packages/vuetify/src/components/VCalendar/__tests__/VCalendar.spec.ts (hunk 7 above)
Comment (johnleider) [reply packages/vuetify/src/components/VCalendar/__tests__/VCalendar.spec.ts:112]: Agreed.

**Comment by genofire** [comment]
I miss documentations of it - will it come?
